streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutionMetadata.java

```diff
@@ -44,13 +44,15 @@ public class TaskExecutionMetadata {
     private final Collection<Task> successfullyProcessed = new HashSet<>();
     // map of topologies experiencing errors/currently under backoff
     private final ConcurrentHashMap<String, NamedTopologyMetadata> topologyNameToErrorMetadata = new ConcurrentHashMap<>();
+    private final Logger log;
 
     public TaskExecutionMetadata(final Set<String> allTopologyNames,
                                  final Set<String> pausedTopologies,
                                  final ProcessingMode processingMode) {
         this.hasNamedTopologies = !(allTopologyNames.size() == 1 && allTopologyNames.contains(UNNAMED_TOPOLOGY));
         this.pausedTopologies = pausedTopologies;
         this.processingMode = processingMode;
+        this.log = new LogContext("[TaskExecutionMetadata]").logger(TaskExecutionMetadata.class);
```

Comment:
Why do we pass the class name as log prefix, given that we set the class via `logger(...)` anyway?